server/app/socketService.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable consistent-return */
+const SocketIo = require('socket.io');
+const SocketClient = require('socket.io-client');
+const httpServer = require('http').createServer();
+const User = require('./models/User');
+
+const ENDPOINT = process.env.SOCKET_CHANNEL_ENDPOINT || 'http://localhost';
+const PORT = process.env.SOCKET_CHANNEL_PORT || 3002;
+
+const io = SocketIo();
+const socketServer = {};
+
+/*
+ *  List of connected clients
+ *  {
+ *    [userId]: SocketObject
+ *  }
+ */
+socketServer.clients = {};
+
+socketServer.start = server => {
+  io.attach(server);
+  io.on('connection', socket => {
+    if (process.env.NODE_ENV !== 'production') {
+      console.log(`Socket: new connection from ${socket.id}`);
+    }
+
+    setTimeout(() => {
+      if (!socket.authenticated) return socket.disconnect(true);
+    }, 2000);
```

Comment:
Is there a case where this might be needed? Does the 'authenticate' listener not take care of this?

Reply:
Yes, shouldn't be necessary